### ISSUE TITLE
CMSIS SVD

### ISSUE THREAD

**Issue by raffi-g** [esynr3z/corsair#12]
Hi! First of all, thanks for the great work so far!

I would like to add a generator for CMSIS SVD files (a register description format for debugging, see [here](https://www.keil.com/pack/doc/cmsis/SVD/html/svd_Format_pg.html)).

I've already done most of the work. The only thing is, that SVD wants to know, in which peripherals the registers are, e.g.

    <peripherals>
        <peripheral>
            
            <registers>
                
            </registers>
        </peripheral>
        <peripheral>
            
        </peripheral>
    </peripherals>

At the moment this information is not provided in the register map. In order to minimize the impact on the current defintion and also be backwards compatible, I would suggest to just add an entry "peripheral" to each register, like so:

    {
            "name": "CSR_NAME",
            "description": "",
            "address": "0x20",
            "peripheral": "my_peripheral",
            "bitfields": []
    }

What's your opinion? Or is this not wished at all? Then I would just leave it on my fork.

**Comment by esynr3z** [comment]
Hello! 

Of course I'm interested in SVD generator! CMSIS SVD was one of the sources of inspiration for me.

I see that you've done plenty of work already. And I'm planning to do a big refactoring, so I'm sure there will be much elegant way to integrate support of several peripherals. For example, to add several register maps to `csrconfig`. I think it is much convenient to place register map of each peripheral in a separate file.

**Comment by raffi-g** [comment]
Hi
Goot to hear. If I can support you in any way, just let me know.

**Comment by esynr3z** [comment]
This is nice feature that has to be done, eventually. However, for me now refactoring and 2.0 is the milestone, so this feature take place only after that.

Moreover, I have plans to add support of multiple register maps in one file, so SVD will get "peripheral" out of the box.
